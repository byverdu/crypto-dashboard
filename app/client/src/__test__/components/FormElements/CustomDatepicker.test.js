import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CustomDatePicker from '../../../components/FormElements/CustomDatepicker';

const formData = require( '../../../config/data' ).dateFields;

let wrapped;
const component = <CustomDatePicker date="Thu Oct 18 2018 00:00:00 GMT+0100 (British Summer Time)" handleChangeDate={() => {}} formData={formData} />;

beforeEach(() => {
  wrapped = shallow( component );
});

describe( 'CustomDatePicker component', () => {
  it( 'renders correctly', () => {
    const tree = renderer
      .create( component )
      .toJSON();

    expect( tree ).toMatchSnapshot();
  });

  it( 'should have a "handleChangeDate" prop', () => {
    expect( wrapped.props()).toHaveProperty( 'handleChangeDate' );
    expect( typeof wrapped.props().handleChangeDate ).toEqual( 'function' );
  });

  it( 'should have a "date" prop', () => {
    expect( wrapped.props()).toHaveProperty( 'date' );
    expect( wrapped.props().date ).toEqual( 'Thu Oct 18 2018 00:00:00 GMT+0100 (British Summer Time)' );
  });

  it( 'should have a "formData" prop', () => {
    expect( wrapped.props()).toHaveProperty( 'formData' );
    expect( wrapped.props().formData.name ).toEqual( 'dateCrypto' );
  });
});
