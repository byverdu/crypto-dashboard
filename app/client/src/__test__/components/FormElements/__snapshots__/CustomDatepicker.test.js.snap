// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDatePicker component renders correctly 1`] = `
<fieldset
  className="MuiFormControl-root-1"
>
  <legend
    className="MuiFormLabel-root-5 MuiFormLabel-required-10"
  >
    Date Crypto Purchase
    <span
      className="MuiFormLabel-asterisk-11"
    >
       *
    </span>
  </legend>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-1"
    onClick={[Function]}
    onError={undefined}
    onKeyPress={[Function]}
  >
    <label
      className="MuiFormLabel-root-5 MuiFormLabel-filled-9 MuiFormLabel-required-10 MuiInputLabel-root-12 MuiInputLabel-formControl-13 MuiInputLabel-animated-16 MuiInputLabel-shrink-15 MuiInputLabel-outlined-18"
      data-shrink={true}
      htmlFor={undefined}
    >
      Date Crypto Purchase
      <span
        className="MuiFormLabel-asterisk-11"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-32 MuiOutlinedInput-root-19 MuiInputBase-formControl-33 MuiInputBase-adornedEnd-37 MuiOutlinedInput-adornedEnd-23"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="MuiPrivateNotchedOutline-root-49 MuiOutlinedInput-notchedOutline-26"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="MuiPrivateNotchedOutline-legend-50"
          style={
            Object {
              "width": 0,
            }
          }
        />
      </fieldset>
      <input
        aria-invalid={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInputBase-input-42 MuiOutlinedInput-input-27 MuiInputBase-inputAdornedEnd-48 MuiOutlinedInput-inputAdornedEnd-31"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        mask={undefined}
        name="dateCrypto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        pipe={undefined}
        placeholder={undefined}
        readOnly={false}
        required={true}
        rows={undefined}
        type="text"
        value="October 18th"
      />
      <div
        className="MuiInputAdornment-root-54 MuiInputAdornment-positionEnd-57"
      >
        <button
          className="MuiButtonBase-root-64 MuiIconButton-root-58"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-63"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-67"
            >
               
              event
               
            </span>
          </span>
          <span
            className="MuiTouchRipple-root-97"
          />
        </button>
      </div>
    </div>
    
  </div>
</fieldset>
`;
